### PR TITLE
Netfx build

### PR DESCRIPTION
**Pull request by ericstj**
With these changes I am able to successfully build for TargetGroup=netfx

/cc @weshaggard @tarekgh @joperezr

### CODE REVIEW

File: external/netstandard/netstandard.depproj [hunk 1]
```diff
@@ -23,6 +30,27 @@
       </Reference>
     </ItemGroup>
   </Target>
+
+  <Target Name="CopyToAdditionalOutputDirectories"
```
Comment (tarekgh) [inline external/netstandard/netstandard.depproj]: does it make sense to move this traget to one of the targets files? I am seeing this more generic Target.

Comment (ericstj) [inline external/netstandard/netstandard.depproj]: It needs to be done better and put into binplace.targets.  I didn't spend the time to solve that problem so I didn't centralize it.

File: src/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.csproj [hunk 2]
```diff
@@ -20,11 +20,10 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='net46-Windows_NT-Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='net46-Windows_NT-Release|AnyCPU'" />
   <Import Project="System.Net.Http.WinHttpHandler.msbuild" Condition="'$(TargetsWindows)' == 'true'" />
-  <ItemGroup Condition="'$(TargetGroup)' == 'net46'">
+  <ItemGroup Condition="'$(TargetGroup)' == 'net46' OR '$(TargetGroup)' == 'netfx'">
```
Comment (tarekgh) [inline src/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.csproj]: do we need to keep net46 for this project? or netfx should cover it?

File: src/System.Runtime.InteropServices.RuntimeInformation/src/System.Runtime.InteropServices.RuntimeInformation.csproj [hunk 3]
```diff
@@ -6,7 +6,7 @@
     <RootNamespace>System.Runtime.InteropServices</RootNamespace>
     <AssemblyName>System.Runtime.InteropServices.RuntimeInformation</AssemblyName>
     <ProjectGuid>{F9DF2357-81B4-4317-908E-512DA9395583}</ProjectGuid>
-    <DefineConstants Condition="'$(TargetGroup)'=='net45'">net45</DefineConstants>
+    <DefineConstants Condition="'$(TargetGroup)'=='netfx' OR '$(TargetGroup)'=='netfx'">netfx</DefineConstants>
```
Comment (tarekgh) [inline src/System.Runtime.InteropServices.RuntimeInformation/src/System.Runtime.InteropServices.RuntimeInformation.csproj]: did you mean net45 in the condition?

File: src/System.Runtime.Serialization.Primitives/src/Configurations.props [hunk 4]
```diff
@@ -4,7 +4,7 @@
     <BuildConfigurations>
       netcore50aot-Windows_NT;
       uap101aot-Windows_NT;
-      net463-Windows_NT;
+      netfx-Windows_NT;
```
Comment (tarekgh) [inline src/System.Runtime.Serialization.Primitives/src/Configurations.props]: why we are specifying the OS configuration here and not in other configurations files which has netfx?

File: src/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.csproj (hunk 2 above)
Comment (ericstj) [inline src/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.csproj]: I kept net46 because this is in a package and specifically wants to target 4.6 with a live build.

File: src/System.Runtime.InteropServices.RuntimeInformation/src/System.Runtime.InteropServices.RuntimeInformation.csproj (hunk 3 above)
Comment (ericstj) [inline src/System.Runtime.InteropServices.RuntimeInformation/src/System.Runtime.InteropServices.RuntimeInformation.csproj]: Yep, will fix.

File: src/System.Runtime.Serialization.Primitives/src/Configurations.props (hunk 4 above)
Comment (ericstj) [inline src/System.Runtime.Serialization.Primitives/src/Configurations.props]: Because it was specified before.  The only significant difference for netfx is that the OSGroup will be set to Windows_NT, as well as TargetsWindows will be set to true.  If these properties are used by the project the it should set Windows_NT.

File: src/System.Runtime.Serialization.Xml/src/System.Runtime.Serialization.Xml.csproj [hunk 5]
```diff
@@ -6,22 +6,24 @@
...
     <ProjectReference Include="$(SourceDir)System.Private.DataContractSerialization\src\System.Private.DataContractSerialization.csproj" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetGroup)' == 'net463'">
-    <TargetingPackReference Include="System" />
-    <TargetingPackReference Include="System.Runtime.Serialization" />
+  <ItemGroup Condition="'$(TargetGroup)' == 'netfx'">
+    <Reference Include="mscorlib" />
+    <Reference Include="System" />
+    <Reference Include="System.Runtime.Serialization" />
+    <ProjectReference Include="$(SourceDir)System.Runtime.Serialization.Primitives\src\System.Runtime.Serialization.Primitives.csproj" />
```
Comment (tarekgh) [inline src/System.Runtime.Serialization.Xml/src/System.Runtime.Serialization.Xml.csproj]: we don't include System.Runtime.Serialization.Primitives in the runtime folder? I am just wondering why we do project reference instead of just regular reference

File: src/System.Security.Cryptography.Cng/ref/System.Security.Cryptography.Cng.cs [hunk 6]
```diff
@@ -276,17 +276,23 @@ public sealed partial class ECDsaCng : System.Security.Cryptography.ECDsa
         public ECDsaCng() { }
         public ECDsaCng(int keySize) { }
         public ECDsaCng(System.Security.Cryptography.CngKey key) { }
+#if !netfx // types missing from netfx targeting pack
```
Comment (tarekgh) [inline src/System.Security.Cryptography.Cng/ref/System.Security.Cryptography.Cng.cs]: @weshaggard any idea if the desktop targeting pack is going to include these at some point?

File: src/System.Runtime.Serialization.Xml/src/System.Runtime.Serialization.Xml.csproj (hunk 5 above)
Comment (ericstj) [inline src/System.Runtime.Serialization.Xml/src/System.Runtime.Serialization.Xml.csproj]: Because we need the implementation assembly in order to unify types.  Regular reference would get the ref folder version which doesn't have the type forwards.  Runtime reference would be a race condition.  Project reference is same binary as runtime reference without the race condition.

There is a generic problem I was dealing with here and in other projects was what to do when we need to build a netfx facade for an assembly that is in NETStandard.Library.  For these we don't have a netstandard contract build.  I'm creating a netfx build but its kinda busted since we end up with a mix of facades and reference assemblies in the ref folder for netfx.  We have to have a reference assembly in order to produce the facade from source, while at the same time all the files from NETStandard.Library package in ref folder are facades.

File: external/netstandard/netstandard.depproj [hunk 7]
```diff
@@ -6,14 +6,21 @@
     <NugetRuntimeIdentifier>None</NugetRuntimeIdentifier>
     <OutputPath>$(RefRootPath)/$(TargetGroup)</OutputPath>
     <ProjectJsonTemplate Condition="$(NuGetTargetMoniker.Contains('Version=v1.'))">$(MSBuildProjectDirectory)\netstandard1.x\project.json.template</ProjectJsonTemplate>
+    <_NETStandardTFMFolder Condition="'$(NuGetTargetMoniker)' == '.NETStandard,Version=v2.0'">netstandard2.0</_NETStandardTFMFolder>
+    <_NETStandardTFMFolder Condition="$(NuGetTargetMoniker.StartsWith('.NETFramework'))">net461</_NETStandardTFMFolder>
```
Comment (ericstj) [inline external/netstandard/netstandard.depproj]: We need to exclude the facades for the cases where we need to build a newer copy from the corefx repo (eg: system.data.common, system.numerics.vectors, etc).  /cc @weshaggard

**Comment by tarekgh** [comment]
Thanks @ericstj LGTM

File: src/System.Runtime.Serialization.Primitives/src/Configurations.props (hunk 4 above)
Comment (weshaggard) [inline src/System.Runtime.Serialization.Primitives/src/Configurations.props]: I do think we should always specify Windows_NT for all our netfx configuration.

File: Tools-Override/FrameworkTargeting.targets [hunk 8]
```diff
@@ -72,6 +72,16 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="MarkReferencePrivateFalse"
+          BeforeTargets="ResolveAssemblyReferences">
+    <ItemGroup>
+      <!-- Simple name references will be resolved from the targeting pack folders and should never be copied to outpu -->
```
Comment (weshaggard) [inline Tools-Override/FrameworkTargeting.targets]: nit: outpu -> output

File: external/netfx/netfx.depproj [hunk 9]
```diff
@@ -6,15 +6,16 @@
     <NugetRuntimeIdentifier>None</NugetRuntimeIdentifier>
     <OutputPath>$(RefRootPath)/$(TargetGroup)</OutputPath>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(TargetGroup)' == 'net463'">
+  <PropertyGroup Condition="'$(NuGetTargetMoniker)' == '.NETFramework,Version=v4.6.3'">
```
Comment (weshaggard) [inline external/netfx/netfx.depproj]: Why did you need to switch to condition on the NugetTargetMoniker?

File: external/netstandard/netstandard.depproj (hunk 7 above)
Comment (weshaggard) [inline external/netstandard/netstandard.depproj]: I don't see where you are filtering those? Can't we get a way with letting the corefx build overwrite them?

File: external/netfx/netfx.depproj (hunk 9 above)
Comment (tarekgh) [inline external/netfx/netfx.depproj]: I think to include both netfx and net463.

File: src/System.Runtime.Serialization.Primitives/src/Configurations.props (hunk 4 above)
Comment (ericstj) [inline src/System.Runtime.Serialization.Primitives/src/Configurations.props]: Arbitrarily?  Why would we do that?

File: external/netstandard/netstandard.depproj (hunk 7 above)
Comment (ericstj) [inline external/netstandard/netstandard.depproj]: I'm not filtering yet.  Right now I am just overwriting them but was planning to filter.  I can continue to overwrite if you think its ok.

File: external/netfx/netfx.depproj (hunk 9 above)
Comment (ericstj) [inline external/netfx/netfx.depproj]: Yep.

File: src/System.Runtime.Serialization.Primitives/src/Configurations.props (hunk 4 above)
Comment (weshaggard) [inline src/System.Runtime.Serialization.Primitives/src/Configurations.props]: Because I want to be sure that they only build for the windows build virtual. netfx-Unix makes really no sense and so if folks want to target netfx then they should either be on windows or specify OSGroup=Windows_NT.

File: external/netstandard/netstandard.depproj (hunk 7 above)
Comment (weshaggard) [inline external/netstandard/netstandard.depproj]: Assuming things are correctly staged, which I believe they are I think I would prefer we overwrite instead of maintaining a filtering list.

File: src/System.Security.Cryptography.Cng/ref/System.Security.Cryptography.Cng.cs (hunk 6 above)
Comment (weshaggard) [inline src/System.Security.Cryptography.Cng/ref/System.Security.Cryptography.Cng.cs]: I'm sure they will. I'm not entirely sure which version they were added in.

File: external/netstandard/netstandard.depproj (hunk 7 above)
Comment (ericstj) [inline external/netstandard/netstandard.depproj]: Ok, we'll have to make sure that doesn't break incremental build...

File: src/System.Xml.XPath.XDocument/src/Configurations.props [hunk 10]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <BuildConfigurations>
-      net463-Windows_NT;
+      <!-- netfx-Windows_NT; clashes with NETStandard.Library -->
```
Comment (weshaggard) [inline src/System.Xml.XPath.XDocument/src/Configurations.props]: Based on our other comments I assume you will re-enable any of these that we still plan to ship a package for.

File: src/System.Runtime.Serialization.Primitives/src/Configurations.props (hunk 4 above)
Comment (ericstj) [inline src/System.Runtime.Serialization.Primitives/src/Configurations.props]: If you want to block certain verticals from building that's a different story.

I don't think it makes sense to configure every project to be OSGroup specific just so you don't get a binary when you specify a config that *might* be invalid (I don't agree, btw, what about mono?).

Even if we did this for all projects targeting netfx, you'd still get binaries dropped for netstandard configs that don't specify an OSGroup when building for vertical netfx-Unix.

IMHO without any enforcement, you're just forcing folks to type more.

File: src/System.Xml.XPath.XDocument/src/Configurations.props (hunk 10 above)
Comment (ericstj) [inline src/System.Xml.XPath.XDocument/src/Configurations.props]: Yeah, that *WIP* in the title is referring to these.

File: Tools-Override/FrameworkTargeting.targets (hunk 8 above)
Comment (ericstj) [inline Tools-Override/FrameworkTargeting.targets]: fixed

File: src/System.Xml.XPath.XDocument/src/Configurations.props (hunk 10 above)
Comment (ericstj) [inline src/System.Xml.XPath.XDocument/src/Configurations.props]: I enabled most.  Some were no longer the "home" assemblies for these types.  I think we'll need to do an API compat run to decide which types need to be added and in which assembly (eg: ns1.6 home or ns2.0 home).

File: src/System.Runtime.Serialization.Primitives/src/Configurations.props (hunk 4 above)
Comment (ericstj) [inline src/System.Runtime.Serialization.Primitives/src/Configurations.props]: /cc @joperezr who I noticed just added a bunch of stuff for uap-Windows_NT.
